core.go

```diff
@@ -407,6 +417,30 @@ func (c *Core) RemoveLookup(lookup, lookupID, key string) (err error) {
 	return
 }
 
+// Backup will create a full backup of the db into a given directory path
+func (c *Core) Backup(ctx context.Context, dir string, example Value) (err error) {
+	var backup *Core
+	if backup, err = New(c.name, dir, example, c.getRelationships()...); err != nil {
+		return
+	}
+	defer backup.Close()
+
+	if err = c.ForEach(func(key string, val Value) (err error) {
+		if err = backup.Transaction(ctx, func(txn *Transaction) (err error) {
```

Comment:
@classic-massok what if we took a dynamic batch size, then ran batched transactions in groups?

Reply:
Or even a fixed batch size? I just see this taking a very long time.